F21/yzhang120/lower_case.py

```diff
@@ -0,0 +1,43 @@
+import pandas as pd
+
+def lower_general(input_string):
+    return input_string.lower()
+
+def lower_null(input_string):
+    return input_string
+
+def lower_tr_az(input_string):
+    output_string = ''
+    for char in input_string:
+        if char != 'I':
+            output_string += char.lower()
+        else:
+            output_string += u"\u0131"
+    return output_string
+
+def lower_ga(input_string):
+    Irish_vowel = 'AEIOUÁÉÍÓÚ'
+    if input_string[0] in 'nt' and input_string[1] in Irish_vowel:
+        output_string = input_string[0] + '-' + input_string[1:].lower()
+    else:
+        output_string = input_string.lower()
+    return output_string
+
+def get_lowercase(word, language):
```

Comment:
Since this is written using the BCP-47 specification, you can use the first two letters of the language to determine the langauge rules the lowercase function should adhere to. You may want to strip the input first to get rid of any extra white space that may be included. Then you don't have to include every single case for each language.